### ISSUE TITLE
How to set L1 weight decay? 

### ISSUE THREAD

**Issue by backyes** [PaddlePaddle/Paddle#267]
- global setting(...) function call has no L1 weight decay setting, so, how to set L1 weight decay? 
- can set different L1 weight decay for different inputs of same layer in model config file?

**Comment by backyes** [comment]
L1 example:
data = data_layer(name="feature", size=feature_dict_size)
hidden1 = fc_layer(name="hidden1",
                   input=data,
                   size=8,
                   param_attr=ParameterAttribute(l1_rate=0.1,
                                                 sparse_update=True))

L2 example: 
settings(
    batch_size=batch_size,
    learning_rate=0.1,
    learning_method=RMSPropOptimizer(),
    regularization=L2Regularization(1.0),
)
set L1 with parameter level,  while you can set L2 with global and parameter level.

**Comment by emailweixu** [comment]
Should add L1Regularization() for settings(). In fact, the effect of L2Regularization() is to have a default l2_rate for each parameter.

**Comment by backyes** [comment]
@emailweixu  I will follow it. 
* add L1Regularization() 
* more detailed doc about decay rate

In addition, for pure dense local updater, L1 is not supported in SgdLocalUpdater. @emailweixu pls give some more comments on it if feel free.

**Comment by reyoung** [comment]
@backyes Is this issue should associate to some milestone? Should we support it in 2 weeks or 4 weeks?

**Comment by buptzzl** [comment]
@reyoung   L1 正则为啥本地训练不能用啊？这样会影响训练出的模型的效果

**Comment by reyoung** [comment]
@buptzzl Hello, @backyes is working on this issue. Please ask him for progress or just subscribe this issue.

**Comment by backyes** [comment]
For local mode, if the layer set with sparse_update=true, the L1 indeed works. 

```
data = data_layer(name="feature", size=feature_dict_size)
hidden1 = fc_layer(name="hidden1",
                   input=data,
                   size=8,
                   act=LinearActivation(),
                   param_attr=ParameterAttribute(l1_rate=1, sparse_update=True))
```
That works well. @buptzzl 

You can paste your model if you feel free.

**Comment by qingqing01** [comment]
The L1 weight decay can be set in ParameterAttribute: https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/attrs.py#L146 . Close this issue.  If any questions, we will fix it in the new issue.
